src/RA_Defs.h

```diff
@@ -52,45 +55,58 @@
+#include <rapidjson/document.h>
+#include <rapidjson/writer.h>
+#include <rapidjson/filestream.h>
+#include <rapidjson/error/en.h>
 using namespace rapidjson;
 extern GetParseErrorFunc GetJSONParseErrorStr;
 #pragma warning(pop)
 #endif	//RA_EXPORTS
 
 
-#define RA_KEYS_DLL						"RA_Keys.dll"
-#define RA_PREFERENCES_FILENAME_PREFIX	"RAPrefs_"
-#define RA_UNKNOWN_BADGE_IMAGE_URI		"00000"
+inline constexpr auto RA_KEYS_DLL                    = TEXT("RA_Keys.dll");
+inline constexpr auto RA_PREFERENCES_FILENAME_PREFIX = TEXT("RAPrefs_");
+inline constexpr auto RA_UNKNOWN_BADGE_IMAGE_URI	 = TEXT("00000");
 
-#define RA_DIR_OVERLAY					".\\Overlay\\"
-#define RA_DIR_BASE						".\\RACache\\"
-#define RA_DIR_DATA						RA_DIR_BASE##"Data\\"
-#define RA_DIR_BADGE					RA_DIR_BASE##"Badge\\"
-#define RA_DIR_USERPIC					RA_DIR_BASE##"UserPic\\"
-#define RA_DIR_BOOKMARKS				RA_DIR_BASE##"Bookmarks\\"
+inline constexpr auto RA_DIR_OVERLAY                 = TEXT(".\\Overlay\\");
+inline constexpr auto RA_DIR_BASE	                 = TEXT(".\\RACache\\");
 
-#define RA_GAME_HASH_FILENAME			RA_DIR_DATA##"gamehashlibrary.txt"
-#define RA_GAME_LIST_FILENAME			RA_DIR_DATA##"gametitles.txt"
-#define RA_MY_PROGRESS_FILENAME			RA_DIR_DATA##"myprogress.txt"
-#define RA_MY_GAME_LIBRARY_FILENAME		RA_DIR_DATA##"mygamelibrary.txt"
+inline constexpr auto RA_DIR_DATA                    = TEXT(".\\RACache\\Data\\");
+inline constexpr auto RA_DIR_BADGE                   = TEXT(".\\RACache\\Badge\\");
+inline constexpr auto RA_DIR_USERPIC                 = TEXT(".\\RACache\\UserPic\\");
+inline constexpr auto RA_DIR_BOOKMARKS               = TEXT(".\\RACache\\Bookmarks\\");
 
-#define RA_OVERLAY_BG_FILENAME			RA_DIR_OVERLAY##"overlayBG.png"
-#define RA_NEWS_FILENAME				RA_DIR_DATA##"ra_news.txt"
-#define RA_TITLES_FILENAME				RA_DIR_DATA##"gametitles.txt"
-#define RA_LOG_FILENAME					RA_DIR_DATA##"RALog.txt"
+inline constexpr auto RA_GAME_HASH_FILENAME	         = TEXT(".\\RACache\\Data\\gamehashlibrary.txt");
+inline constexpr auto RA_GAME_LIST_FILENAME	         = TEXT(".\\RACache\\Data\\gametitles.txt");
+inline constexpr auto RA_MY_PROGRESS_FILENAME        = TEXT(".\\RACache\\Data\\myprogress.txt");
+inline constexpr auto RA_MY_GAME_LIBRARY_FILENAME    = TEXT(".\\RACache\\Data\\mygamelibrary.txt");
 
+inline constexpr auto RA_OVERLAY_BG_FILENAME         = TEXT(".\\Overlay\\overlayBG.png");
+inline constexpr auto RA_NEWS_FILENAME               = TEXT(".\\RACache\\Data\\ra_news.txt");
+inline constexpr auto RA_TITLES_FILENAME             = TEXT(".\\RACache\\Data\\gametitles.txt");
+inline constexpr auto RA_LOG_FILENAME                = TEXT(".\\RACache\\Data\\RALog.txt");
 
-#define RA_HOST_URL						"retroachievements.org"
-#define RA_HOST_IMG_URL					"i.retroachievements.org"
+inline constexpr auto RA_HOST_URL                    = TEXT("retroachievements.org");
+inline constexpr auto RA_HOST_IMG_URL                = TEXT("i.retroachievements.org");
+
```

Comment:
First off these are aligned, before the change they didn't look aligned to begin with.

We could make them constant expressions but not like before. I put TEXT for good measure because the Windows API flips between multibyte and byte string.

Reply:
alignment issues are due to tabs v. spaces. I've noticed that the reformatter doesn't change them in the middle of lines. In this particular block of code, there's a tab immediately after `RA_GAME_HASH_FILENAME` and `RA_GAME_LIST_FILENAME` (try highlighting the single space after each).

Reply:
I think it would be easier for me to just convert tabs to spaces in Notepad++, it would show up as lots of lines. That might be better another PR, but for now I'll just look at what I'm changing.